pkg/gardenlet/controller/shoot/shoot_control_prepare_migration.go

```diff
@@ -0,0 +1,288 @@
+		shoot.Status.LastErrors = nil
+		shoot.Status.LastOperation = &gardencorev1beta1.LastOperation{
+			Type:           gardencorev1beta1.LastOperationTypeMigrate,
+			State:          gardencorev1beta1.LastOperationStateSucceeded,
+			Progress:       100,
+			Description:    successDescription,
+			LastUpdateTime: metav1.Now(),
+		}
+		return shoot, nil
+	}
+}
+
+func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation) *gardencorev1beta1helper.WrappedLastErrors {
+	var (
+		botanist        *botanistpkg.Botanist
+		err             error
+		dnsEnabled      = !o.Shoot.DisableDNS
+		shootHibernated = isShootHibernated(o.Shoot.Info.Spec.Hibernation)
+		tasksWithErrors []string
+	)
+
+	for _, lastError := range o.Shoot.Info.Status.LastErrors {
+		if lastError.TaskID != nil {
+			tasksWithErrors = append(tasksWithErrors, *lastError.TaskID)
+		}
+	}
+
+	errorContext := utilerrors.NewErrorContext("Shoot's control plane preparation for migration", tasksWithErrors)
+
+	err = utilerrors.HandleErrors(errorContext,
+		func(errorID string) error {
+			o.CleanShootTaskError(context.TODO(), errorID)
+			return nil
+		},
+		nil,
+		utilerrors.ToExecute("Create botanist", func() error {
+			return retryutils.UntilTimeout(context.TODO(), 10*time.Second, 10*time.Minute, func(context.Context) (done bool, err error) {
+				botanist, err = botanistpkg.New(o)
+				if err != nil {
+					return retryutils.MinorError(err)
+				}
+				return retryutils.Ok()
+			})
+		}),
+	)
+
+	if err != nil {
+		return gardencorev1beta1helper.NewWrappedLastErrors(gardencorev1beta1helper.FormatLastErrDescription(err), err)
+	}
+
+	var (
+		defaultTimeout  = 10 * time.Minute
+		defaultInterval = 5 * time.Second
+
+		g = flow.NewGraph("Shoot's control plane preparation for migration")
+
+		deploySecrets = g.Add(flow.Task{
+			Name: "Deploying Shoot certificates / keys",
+			Fn:   flow.TaskFn(botanist.DeploySecrets),
+		})
+		deployBackupEntryInGarden = g.Add(flow.Task{
+			Name: "Deploying backup entry",
+			Fn:   flow.TaskFn(botanist.DeployBackupEntryInGarden),
+		})
+		wailtUntilBackupEntryInGardenReconciled = g.Add(flow.Task{
+			Name:         "Waiting until the backup entry has been reconciled",
+			Fn:           flow.TaskFn(botanist.WaitUntilBackupEntryInGardenReconciled),
+			Dependencies: flow.NewTaskIDs(deployBackupEntryInGarden),
+		})
+		deployETCD = g.Add(flow.Task{
+			Name:         "Deploying main and events etcd",
+			Fn:           flow.TaskFn(botanist.DeployETCD).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(deploySecrets, wailtUntilBackupEntryInGardenReconciled),
+		})
+		waitUntilEtcdReady = g.Add(flow.Task{
+			Name:         "Waiting until main and event etcd report readiness",
+			Fn:           flow.TaskFn(botanist.WaitUntilEtcdReady),
+			Dependencies: flow.NewTaskIDs(deployETCD),
+		})
+		prepareControlPlaneDeployments = g.Add(flow.Task{
+			Name:         "Stop Kubernetes API Server in Shoot's namespace",
```

Comment:
The task name is a bit different from the function name. Can we either rename the function or the task name?